File: LHDS.AdminPortal.Web/ClientApp/src/models/suppliers/supplier.ts
import { Guid } from 'guid-typescript';

export class Supplier {
    public id: Guid;
    public name?: string;
    public friendlyName?: string;
    public description?: string;
    public isIngestionTracked?: boolean;
    public canDecryptIngestionTracking?: boolean;
    public createdBy?: string;
    public createdDate?: Date;
    public updatedBy?: string;
    public updatedDate?: Date;

    constructor(supplier: any) {
        this.id = supplier.id ? Guid.parse(supplier.id) : Guid.parse(Guid.EMPTY);
        this.name = supplier.name;
        this.friendlyName = supplier.friendlyName;
        this.description = supplier.description;
        this.isIngestionTracked = supplier.isIngestionTracked;
        this.canDecryptIngestionTracking = supplier.canDecryptIngestionTracking === true ? true : false;
        this.createdBy = supplier.createdBy;
        this.createdDate = new Date(supplier.createdDate);
        this.updatedBy = supplier.updatedBy;
        this.updatedDate = new Date(supplier.updatedDate);
    }
}

File: LHDS.ConfigImportExportTool/Models/Foundations/Suppliers/Supplier.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;
using System.Collections.Generic;
using LHDS.ConfigImportExportTool.Models.Bases;
using LHDS.ConfigImportExportTool.Models.Foundations.Datasets;
using Microsoft.AspNetCore.Mvc.ModelBinding;

namespace LHDS.ConfigImportExportTool.Models.Foundations.Suppliers
{
    public class Supplier : IKey, IAudit
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string FriendlyName { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public bool canRelandIngestionTracking { get; set; }
        public bool CanDecryptIngestionTracking { get; set; }
        public bool CanDownloadIngestionTracking { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
        public DateTimeOffset UpdatedDate { get; set; }

        [BindNever]
        public List<DataSet> DataSets { get; set; } = new List<DataSet>();
    }
}
